library-api/src/main/java/cz/muni/fi/pa165/library/facade/LoanFacade.java

```diff
@@ -7,13 +7,13 @@
 import java.util.List;
 
 /** @author Martin Páleník 359817 */
-public interface SingleLoanFacade {
+public interface LoanFacade {
 
-    /** The time of loan will be set to now() */
     public void borrowBook(SingleLoanDTO singleLoanDto);
+    public void borrowBooks(List<SingleLoanDTO> loans);
 
```

Comment:
cant we have just 
`public void createLoan(LoanDTO loan)` ? LoanDTO will be constructed on FE.
But we need to be able to return just a single book. So we still need some SingleLoanFacade with method returnBook(SingleLoanDto singleLoan); Use case: viewLoansForCustomer shows all SingleLoans, button "return book" calls this (returnBook) facade method over API

Reply:
> cant we have just `public void createLoan(LoanDTO loan)` ?

LoanDTO does not exist, did you mean to suggest renaming `SingleLoanDTO` to `LoanDTO`? The problem is it would not match the corresponding DAO. 

I can rename `borrowBook` to `createLoan` if you want. 

> LoanDTO will be constructed on FE.

What does FE stand for?

> But we need to be able to return just a single book. So we still need some SingleLoanFacade with method returnBook(SingleLoanDto singleLoan);

The responsibility of "SingleLoanFacade" can be attributed to LoanFacade. We don't need to map each service/repository into a facade. The sample project also has "Order" handling multiple "OrderItem" while only having one facade for handling all orders. There is simply no need for two facades to cover the same functionality - handling loans. 

> Use case: viewLoansForCustomer shows all SingleLoans, button "return book" calls this (returnBook) facade method over API

There is already a method `getLoansForUser` in the LoanFacade that does exactly that. Is there something wrong with it?

Reply:
FE is frontend

Reply:
I suggested CREATING LoanDto

Reply:
Yes, rename it please

Reply:
List<SingleLoanDTO> loans is effectively LoanDto

Reply:
How is returning a single item handled?

Reply:
To remind you: SingleLoan is an association between book and user.
Loan is container of SingleLoans.
I want to be able to create Loan given LoanDto, which contains multiple SingleLoanDtos inside.
I want to be able to return SingleLoanDto.

Reply:
If you think its prettier to stick it all under LoanFacade, do it this way

Reply:
> How is returning a single item handled?

It is handled by `returnBook(SingleLoanDTO singleLoanDto)`. After renaming `borrowBook` as you requested we would have `createLoan` for borrowing and `returnBook` for returning. I find that rather confusing.

If you meant the getter, then `SingleLoanDTO getSingleLoanById(long id)` returns a single item by id.

Reply:
> How is returning a single item handled?

It is handled by `returnBook(SingleLoanDTO singleLoanDto)`. After renaming `borrowBook` to `createLoan` as you suggest, we would have `createLoan` for borrowing and `returnBook` for returning. I find that rather confusing. Since `returnLoan` does not make sense, wouldn't it be better to keep it the way it is?

If you meant the getter, then `SingleLoanDTO getSingleLoanById(long id)` returns a single item.

> I want to be able to return SingleLoanDto.

Then `getSingleLoanById` is probably what you want.

Reply:
Please check the `createLoan` and `createLoans`. Is it all right now?